library/std/src/os/unix/net/ancillary.rs

```diff
@@ -625,6 +726,51 @@ impl<'a> SocketAncillary<'a> {
         )
     }
 
+    /// Add TTL to the ancillary data.
+    ///
+    /// The function returns `true` if there was enough space in the buffer.
+    /// If there was not enough space then no file descriptors was appended.
+    /// Technically, that means this operation adds a control message with the level `IPPROTO_IP`
+    /// and type `IP_TTL`.
+    ///
+    /// # Example
+    /// ```no_run
+    /// #![feature(unix_socket_ancillary_data)]
+    /// use std::io::IoSlice;
+    /// use std::net::UdpSocket;
+    /// use std::os::unix::net::SocketAncillary;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let socket = UdpSocket::bind("127.0.0.1:34254").expect("couldn't bind to address");
+    ///     socket.connect("127.0.0.1:41203").expect("couldn't connect to address");
+    ///     let buf1 = [1; 8];
+    ///     let buf2 = [2; 16];
+    ///     let buf3 = [3; 8];
+    ///     let bufs = &[
+    ///         IoSlice::new(&buf1),
+    ///         IoSlice::new(&buf2),
+    ///         IoSlice::new(&buf3),
+    ///     ][..];
+    ///     let mut ancillary_buffer = [0; 128];
+    ///     let mut ancillary = SocketAncillary::new(&mut ancillary_buffer[..]);
+    ///     ancillary.add_ipttl(20);
+    ///     socket.send_vectored_with_ancillary(bufs, &mut ancillary)
+    ///         .expect("send_vectored_with_ancillary function failed");
+    ///     Ok(())
+    /// }
+    /// ```
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub fn add_ipttl(&mut self, ttl: u8) -> bool {
```

Comment:
Does this actually work to set the TTL field for sent packets? I couldn't find any documentation that mentions the user giving `IP_TTL` control messages to the kernel, only for the kernel giving it to the user.

As far as I understood it, you need to set the IP_TTL socket option to actually change the field for outgoing packets. That also kinda makes sense, because there is no one-on-one mapping of `sendmsg` calls to outgoing IP packets.

Reply:
`IP_TTL` should work with `sendmsg`, else this [test](https://github.com/rust-lang/rust/pull/82858/commits/6a1169c4db52ff337d3e041f662bf846367d705a#diff-dcbf9cec3d8d653c4dab2857df4de3a44f722ac660b9a380da9f9cbceb094276R418) would be failing.

Reply:
Interesting. Although I would still feel a lot better if this was a documented feature.

Reply:
Otherwise, it would pretty inefficient, If you have to call `setsockopt` each time you want to send a package with a specific TTL.